examples/usvg_viewer/src/main.rs

```diff
@@ -0,0 +1,212 @@
+mod asset;
+mod render;
+
+use anyhow::Result;
+use asset::ASSETS;
+use byte_unit::Byte;
+use clap::Parser;
+use dialoguer::Confirm;
+use render::render_svg;
+use std::path::PathBuf;
+use std::time::Instant;
+use vello::{
+    kurbo::{Affine, Vec2},
+    util::RenderContext,
+    Renderer, Scene, SceneBuilder,
+};
+use winit::{
+    dpi::LogicalSize,
+    event_loop::EventLoop,
+    window::{Window, WindowBuilder},
+};
+
+#[derive(Parser, Debug)]
+#[command(about, long_about = None)]
+struct Args {
+    /// Input files for rendering. Will use builtin SVGs if empty.
+    files: Vec<PathBuf>,
+}
+
+// Check if all the known assets have been downloaded.
+// If some haven't been downloaded (or if the checksums don't match), find
+// their combined size and licenses. Ask the user if they want to download
+// the SVG files.
+// If yes, download the files and return normally.
+// If no, exit with status code -1
+fn fetch_missing_assets() -> Result<()> {
+    let missing_assets = ASSETS
+        .iter()
+        .filter(|asset| !asset.fetched())
+        .collect::<Vec<_>>();
+
+    if !missing_assets.is_empty() {
+        let total_size = Byte::from_bytes(missing_assets.iter().map(|asset| asset.size).sum())
+            .get_appropriate_unit(true);
+        let mut licenses: Vec<_> = missing_assets.iter().map(|asset| asset.license).collect();
+        licenses.dedup();
+
+        println!("Some SVG assets are missing. Let me download them for you.");
+        println!(
+            "They'll take up {total_size} and are available under these licenses: {}",
+            licenses.join(", ")
+        );
+
+        if Confirm::new()
```

Comment:
It seems strange to use a CLI interface for a GUI tool, but I don't really have a better suggestion